lib/loan/ui/pages/admin_page/loan_card.dart

```diff
@@ -120,7 +120,7 @@ class LoanCard extends StatelessWidget {
                         color: shouldReturn
                             ? LoanColorConstants.urgentRed
                             : Colors.grey.shade400)),
-                Text(processDate(loan.end),
+                Text(processDate(loan.returned ? loan.returnedDate! : loan.end),
```

Comment:
We should at least check that returnedDate exist before showing it. Even tho, it should never happen...